src/tree/branch.rs

```diff
@@ -34,6 +31,7 @@ use rkyv::{Archive, Deserialize, Serialize};
 )]
 pub struct PoseidonLevel {
     pub(crate) level: [BlsScalar; dusk_hades::WIDTH],
+    // TODO figure out if we can make this smaller
```

Comment:
For the level its definitely possible but maybe not worthy since it will be stored in a cpu register anyway.

If `usize`, then it can be `u32` if the target architecture uses that

Reply:
Ah, the classic conundrum of space vs. speed. The comment shouldn't really have been kept here, but it seemed strange to me that a number that is always between 0 and 5 is represented using a `u64`.